docling_eval/prediction_providers/aws_prediction_provider.py

```diff
@@ -239,6 +239,221 @@ def convert_aws_output_to_docling(
 
                 doc.add_text(label=DocItemLabel.TEXT, text=text_content, prov=prov)
 
+            if block["BlockType"] == "LAYOUT_TITLE":
+                text_content = block.get("Text", "")
+                bbox = self.extract_bbox_from_geometry(block.get("Geometry", {}))
+
+                # Scale normalized coordinates to the page dimensions
+                bbox_obj = BoundingBox(
+                    l=bbox["l"] * width,
+                    t=bbox["t"] * height,
+                    r=bbox["r"] * width,
+                    b=bbox["b"] * height,
+                    coord_origin=CoordOrigin.TOPLEFT,
+                )
+
+                prov = ProvenanceItem(
+                    page_no=page_no,
+                    bbox=bbox_obj,
+                    charspan=(0, len(text_content)),
+                )
+
+                doc.add_title(text=text_content, prov=prov)
+
+            if block["BlockType"] == "LAYOUT_HEADER":
```

Comment:
please use `elif` instead of `if` and make seperate private (starting with `_` subfunctions)